gradebook/app/ui/src/java/org/sakaiproject/tool/gradebook/ui/AssignmentBean.java

```diff
@@ -429,6 +431,22 @@ public String updateAssignment() {
 			Double newPointsPossible = assignment.getPointsPossible();
 			boolean scoresEnteredForAssignment = getGradebookManager().isEnteredAssignmentScores(assignmentId);
 			
+			// Check if there are assignments which send grades to this gradebook item
+			// If yes, we cannot allow to modify the gradebook item's name
+			AssignmentService aService = (AssignmentService)ComponentManager.get("org.sakaiproject.assignment.api.AssignmentService");
+			Iterator iAssignments = aService.getAssignmentsForContext(getGradebookUid());
+			
+			while (iAssignments.hasNext())
+			{
+				org.sakaiproject.assignment.api.Assignment assignment = (org.sakaiproject.assignment.api.Assignment) iAssignments.next();
+				String associateGradebookAssignment = StringUtils.trimToNull(
+						assignment.getProperties().getProperty(AssignmentService.PROP_ASSIGNMENT_ASSOCIATE_GRADEBOOK_ASSIGNMENT));
+				if (originalAssignment.getName().equals(associateGradebookAssignment)) {
+					FacesUtil.addErrorMessage(getLocalizedString("edit_assignment_error_title_assignment") + assignment.getTitle());
+		            return "failure";
+				}
+			}
+			
```

Comment:
This sounds like the assignment tool is sending the grade based on the gradebook item name. Wasn't there a fix for this recently? I think the fix for this should be done in the assignment tool so it uses the gradebook assignment id, rather than title. The gradebook should be able to change the name of a gradebook item without it breaking anything.